### PR TITLE
Task/135 main app bar widget

### PR DESCRIPTION
**Pull request by TomasPetrik**

### CODE REVIEW

File: lib/view/app_bar/sio_app_bar.dart [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+              top: MediaQuery.of(context).viewPadding.top,
+              bottom: PaddingSize.padding20,
+              left: PaddingSize.padding16,
+            ),
+            child: Row(
+              crossAxisAlignment: CrossAxisAlignment.center,
+              children: [
+                Container(
+                  decoration: const BoxDecoration(
+                    color: Colors.white,
```
Comment (jvrastil) [inline lib/view/app_bar/sio_app_bar.dart:59]: can you use colors from `Theme` or at least `DarkMode`? So we don't forget about it when we refactor colors later...

(same goes with other colors in this file)

File: lib/view/app_bar/sio_app_bar.dart [hunk 2]
```diff
@@ -0,0 +1,138 @@
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:gap/gap.dart';
+import 'package:simplio_app/view/themes/constants.dart';
+import 'package:simplio_app/view/themes/simplio_text_styles.dart';
+
+class SioAppBar extends StatelessWidget {
```
Comment (jvrastil) [inline lib/view/app_bar/sio_app_bar.dart:8]: there is an error for smaller screen iphone SE3 (I tried it on iphone 13 pro max and it did not work as well...):
```
══╡ EXCEPTION CAUGHT BY IMAGE RESOURCE SERVICE ╞════════════════════════════════════════════════════
The following assertion was thrown resolving an image codec:
Unable to load asset: assets/icon/profile_avatar_pic.png

When the exception was thrown, this was the stack:
#0      PlatformAssetBundle.load (package:flutter/src/services/asset_bundle.dart:237:7)
<asynchronous suspension>
#1      AssetBundleImageProvider._loadAsync (package:flutter/src/painting/image_provider.dart:658:14)
<asynchronous suspension>

Image provider: AssetImage(bundle: null, name: "assets/icon/profile_avatar_pic.png")
Image key: AssetBundleImageKey(bundle: PlatformAssetBundle#40cb6(), name:
  "assets/icon/profile_avatar_pic.png", scale: 1.0)
════════════════════════════════════════════════════════════════════════════════════════════════════

Another exception was thrown: A RenderFlex overflowed by 177 pixels on the right.
```

image

File: lib/view/app_bar/sio_app_bar.dart [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+                ),
+                const Gap(PaddingSize.padding10),
+                Expanded(
+                  child: Column(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: [
+                      Text(title,
+                          style: SioTextStyles.h4.apply(
+                              color: Theme.of(context).colorScheme.onPrimary)),
```
Comment (jvrastil) [inline lib/view/app_bar/sio_app_bar.dart:93]: if you add comma after `style` it formats much better. (same for next `Text` widget)

File: lib/view/app_bar/sio_app_bar.dart [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+                      if (subtitle != null) ...{
```
Comment (jvrastil) [inline lib/view/app_bar/sio_app_bar.dart:94]: would there be a difference if you use the spread operator on array?

File: lib/view/app_bar/sio_app_bar.dart [hunk 5]
```diff
@@ -0,0 +1,138 @@
+import 'dart:ui';
```
Comment (storkandstars) [inline lib/view/app_bar/sio_app_bar.dart:1]: reusable widgets should be located in the `view/widgets` folder.

File: lib/view/screens/discovery_screen.dart [hunk 6]
```diff
@@ -1,78 +1,76 @@
...
+                          .goNamed(AuthenticatedRouter.inventory);
+                    },
+                    child: SizedBox(
+                      height: 200,
+                      child: Padding(
+                        padding: CommonTheme.paddingAll,
+                        child: Column(
+                          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                          children: [
+                            //Text(context.locale.gamesTabBarLabel),
```
Comment (jvrastil) [inline lib/view/screens/discovery_screen.dart:53]: remove commented code please

**Review by storkandstars — CHANGES_REQUESTED** [review]

File: lib/view/widgets/bottom_tab_bar.dart [hunk 7]
```diff
@@ -31,74 +34,92 @@ class BottomTabBar extends StatelessWidget {
...
+                        child: floatingActionButton!,
+                      ),
+                  ],
+                ),
               ),
-          ],
-        ),
-      ),
-    );
+            )));
```
Comment (jvrastil) [inline lib/view/widgets/bottom_tab_bar.dart:104]: can you make it more compact by adding commas after each `)`?

**Review by jvrastil — CHANGES_REQUESTED** [review]

File: lib/view/app_bar/sio_app_bar.dart (hunk 1 above)
Comment (TomasPetrik) [reply lib/view/app_bar/sio_app_bar.dart:59]: It's good point, but actually this code will be removed in Task/136 branch.

File: lib/view/app_bar/sio_app_bar.dart (hunk 3 above)
Comment (TomasPetrik) [reply lib/view/app_bar/sio_app_bar.dart:93]: Okay.

File: lib/view/app_bar/sio_app_bar.dart (hunk 4 above)
Comment (TomasPetrik) [reply lib/view/app_bar/sio_app_bar.dart:94]: There wouldn't be a difference in final but you are right. Array shoud be used instead.

File: lib/view/widgets/bottom_tab_bar.dart (hunk 7 above)
Comment (TomasPetrik) [reply lib/view/widgets/bottom_tab_bar.dart:104]: Yes, done.

File: lib/view/app_bar/sio_app_bar.dart (hunk 5 above)
Comment (TomasPetrik) [reply lib/view/app_bar/sio_app_bar.dart:1]: Okey. I moved this widget into view/widgets folder.

File: lib/view/app_bar/sio_app_bar.dart (hunk 2 above)
Comment (TomasPetrik) [reply lib/view/app_bar/sio_app_bar.dart:8]: It looks that it this png asset haven't been found so it broken whole app bar.

I changed image assets to be loaded fomr whole icon folder by:
- assets/icon/

It should works now.

**Review by jvrastil — APPROVED** [review]
looks good to me now 👍

File: lib/view/app_bar/sio_app_bar.dart (hunk 5 above)
Comment (TomasPetrik) [reply lib/view/app_bar/sio_app_bar.dart]: Moved all pending widgets into view/widgets folder.

**Review by storkandstars — APPROVED** [review]

**Pull request merged**